### DISCUSSION TITLE
Extension Marketplace

### DISCUSSION THREAD

**Discussion by cupcakearmy** [directus/directus#2820] [Feature Requests]
Hi there, are there any plans for a market/store for extensions? If done correctly, it would help the product grow a lot, as popular extensions would be easily accessible for any running instance.
Right now one has to manually install it in the hosting environment.
Just an idea :)

**Comment by benhaynes** [comment]
Absolutely! We're working on making all Directus extensions installable as files "over the wire". The plan is to have an external marketplace for browsing extensions, as well as an in-app experience for adding interfaces, etc on the fly.

These extensions would first be created by our Core team (for quality/testing reasons), and then we would open it up to external developers — ideally with a profit-share model.

I've included an initial mockup of what the in-app marketplace might look like:

![Marketplace](https://user-images.githubusercontent.com/522079/135106949-dc3542c0-f69d-4b42-a8a8-9dd694d58d1b.png)

**Reply by cupcakearmy** [reply]
Awesome! Sounds perfect ;)

**Reply by Everus** [reply]
It would be great to have ability to enable/disable extensions on per project basis.

**Reply by cupcakearmy** [reply]
I guess that would be easy once the marketplace is there

**Reply by niklasgrewe** [reply]
cool sounds good. I look forward to it 👍

**Reply by Silbad** [reply]
Will there be a boilerplate for the creation of extensions?

**Reply by benhaynes** [reply]
Yes, but examples of each Extension Type are already available in the Docs...

https://docs.directus.io/guides/displays/
https://docs.directus.io/guides/interfaces/
etc...

**Reply by ghost** [reply]
Some news about marketplace? I'm developing some interfaces that could be useful for another projects. I would like to share them with the community.

**Reply by ijpatricio** [reply]
This is HUGE ❤️

**Reply by hermesalvesbr** [reply]
I'm really anxious waiting for this marketplace.

**Comment by AlexZCP** [comment]
Hi!
Is it a way to write somehowy own extensions already?

**Reply by rijkvanzanten** [reply]
Yes, see https://docs.directus.io/extensions/introduction.html

**Reply by AlexZCP** [reply]
thanks

**Reply by pasha1307** [reply]
I have been trying to create Import CSV button with functionality based on the Extension docs and could not. I looked into the core repo and it has a different structure. Is there more detailed tutorial how to create custom interfaces?  Or to include 3-d party vue npms?

**Comment by B-Esmaili** [comment]
It's mid-2021 still nothing is done for the market. I think rather than focusing merely on you'r landing pages you had better think about the marketplace. Without a good marketplace a CMS is good for nothing if you don't mind.

**Reply by benhaynes** [reply]
That's quite the comment. I find it rude, demanding, and uneducated in how our team/organization prioritizes new features. I think you should engage with the community a bit more before posting.

Not sure what you mean by "focusing on landings pages", but our website has been finished for months. In fact, the website clearly states that npm extensions are coming in Q3-2021 (in the next few months).

**Comment by FullStackAlex** [comment]
@benhaynes  
Any updates on the estimated release date for the marketplace?

**Reply by benhaynes** [reply]
Wrapping up the "over the air" installation, then we'll get this merged into the API, then we'll build this into the App GUI, finally, we'll release a standalone public website for this too.

At any of these stages, the Marketplace will "exist", though the experience will improve as things progress. Impossible to give hard-deadlines for this stuff, but the goal is to have it in and working (at least the API) with the release of 9.0.0 in 2-3 weeks.

**Reply by FullStackAlex** [reply]
Sounds great. Would be great if it lands in the 9.0.0 release. I am very eager to jump into developing apps with Directus. Need a solid WordPress replacement very soon to continue growing as a professional developer. But their extension / plugin system (beside the block editor) is a killer feature.

**Reply by FullStackAlex** [reply]
The website is showing for over one year „Coming Soon“.  Unbelievable. Super irresponsible. 

Even the marketplace gets published any time soon, your information policy on this killed all the fuss around Directus for me. Not to mention the wasted time of a couple of weeks on checking Directus out to develop a proper app.

**Reply by benhaynes** [reply]
The website roadmap is a very vague estimate of what we're planning to work on. We're a small open-source organization, so things change based on our resources, availability, and a million other things. I'm sorry you feel this way, but I feel we've been very open with our community around timelines (or lack thereof).

**Reply by FullStackAlex** [reply]
Your, and your downvoter's feeling is wrong.  You and your fanboys are hyping your platform as something as solid and complete. Also as far as I remember you raised this or last year a couple dozens of millions of bucks. "Open-Source" is not an excuse for attracting developers with tech buzz and then not to deliver essential features for ages. People invest valuable (life!!) time just to check out your promises!! Unlike you, I don't get paid by anyone for this.

**Reply by FullStackAlex** [reply]
And for sure you wasn't transparent with your roadmap on marketplace. You keep promising release "next months". Also downvoting critical comments personally is the lowest level :D

**Reply by FullStackAlex** [reply]
I hope to hear some arguments from the downvoting fanboys. Otherwise I have to assume you all are Directus staff members considering how quickly you downvote.

**Reply by FullStackAlex** [reply]
> I don’t even use Directus, just following the development out of interest as it seems like a great product 

That's the problem. Next please

**Reply by ijpatricio** [reply]
Hello everyone,

I just stopped by to say we just invest the amount of our time that we want. we have many options. stop using, move on to another platform, build your own, use a proprietary licensed one, purchasing maintenance support. The list goes on.

Open-source is a blessing, the output we get from it is insanely valuable!

There are no guarantees in open-source. Feel free to start your open-source project. After maintaining a repository and coordinating bug reports, bug solving, feature requests, pull requests, random comments, and plenty more energy and time-consuming activities... you will then know how to appreciate it. On top of that, try to pay bills with an open-source project.

@benhaynes, you are too kind. keep it up. Even though I'm not involved as I would like with Directus, I'm tinkering from time to time, and can see it growing and growing, adding great features! You and the team are doing an amazing job!

THANK YOU

**Reply by u12206050** [reply]
You don't need the market place to use Directus. Neither is it the market place that will determine whether or not Directus succeeds.

**Reply by Nitwel** [reply]
@FullStackAlex 
I can totaly understand your anger and agree with you that it was missleading by Ben to say that the marketplace would come soon.

But the way you expressed your feelings is ignoring the fact that Directus is a small team with just a few Developers that give their best every day in order to add as many features as possible while also making sure that everything is stable and totally free for everybody to use. In order to have a good marketplace we need to first make sure that extensions are easy to build and install **which we are currently working on**. After that, the marketplace can come but that is also not as easy as it sounds. 

I hope I could clarify the situation a bit for you and why it will take so long to bring a marketplace to Directus.

**Reply by FullStackAlex** [reply]
@u12206050 
YOU don't need the market place to use Directus. But I don't understand why you think you need to comment on this here and not on the original post/request. As you can clearly see on the upvotes of the original post there are a bunch of people waiting impatiently for this "feature", that probably will never come because it probably doesn't fit into the profit roadmap anymore.

**Reply by LasseRosenow** [reply]
@FullStackAlex I am not working for Directus and I downvoted your post, because in my opinion your message was way too offensive. I can understand, that you might feel bad about the delay in the roadmap, but you should know how difficult it can be to estimate the necessary time for features in software development.
In general I think the Directus team already learned from it, hence they changed the roadmap and removed the dates.
So I think it is valid to criticise, but please stay respectful. Nobody forced you to use this product and it was your choice to invest time in it. It is not like you pay for it, so you cannot have demands either.

**Reply by LasseRosenow** [reply]
> > I don’t even use Directus, just following the development out of interest as it seems like a great product
> 
> That's the problem. Next please

@FullStackAlex This again also shows your lack of respect. Not even trying to understand what the person is trying to convey. Just "oh he is not using the product so he has no right to have an opinion on my post." I think this is a really simple minded and non constructive way of thinking :disappointed:

**Reply by ghost** [reply]
Okay , @FullStackAlex you need to STFU, first coz you are rude, secondly this is bloating up my email inbox. You just wasted 5min of 200 people , so thats 1000mins wasted , just coz you are getting stuff for free here. You have no job , so go get one and have some mercy on us. Thanks !!!

**Reply by FullStackAlex** [reply]
@ijpatricio 
> stop using, move on to another platform, build your own, use a proprietary licensed one, purchasing maintenance support. The list goes on.

You're just listing all the available options I know of myself. Useless. Or maybe you are just saying indirectly that that's what I have to do INSTEAD of leaving negative comments here? Then you're wrong as well. I think I have any right to express my huge disappointment with Ben's information policy and the fact that once again I got caught by some superficially fancy looking but over hyped technology.  

> Open-source is a blessing, the output we get from it is insanely valuable!

So if I criticize Ben or Directus, then I criticize "open source"?! Obviously wrong again. 

> There are no guarantees in open source. 

That's correct. But it doesn't mean expressing disappointment with some open source project is forbidden or unethical or illegitim per se. Well, if Ben doesn't want to hear to any critics, than hat's his right as well. It's still his repository. But then just be transparent about it: we don't give any f**k about the needs of the users, we have the following profit-oriented agenda, which we target first, anything else - we don't care about it. Then for sure, you will not read such comments as mine hire. But then you surely will have far less hype around your product. 

>  Feel free to start your open-source project. 

I might even. Still no solid reason to downvote my critics. Ben and his fanboys are promising a great alternative to any other CMS but do not provide a reasonable way to extend it. That should be addressed in any hyped video and blog post. Ben is promising something but fails to deliver. Get it! I fall for it and now I am expressing my legit disappointment about it. No need to tell me to start my own CMS. 

> After maintaining a repository and coordinating bug reports, bug solving, feature requests, pull requests, random comments, and plenty more energy and time-consuming activities... you will then know how to appreciate it. On top of that, try to pay bills with an open-source project.

I don't argue with that. But if you promise something - deliver! People invest weeks, months and some maybe even years in your platform. Take responsibility. Open source is not a plane source of free labour or a free advertising channel for your cloud services. 

> Even though I'm not involved as I would like with Directus, I'm tinkering from time to time, and can see it growing and growing, adding great features! You and the team are doing an amazing job!

lol, again someone who hasn't try to build something reasonable with Directust but feels entitled to educate me....

**Reply by rijkvanzanten** [reply]
Just so we're all hatefully discussing the same thing here:

The _Extension Marketplace_ is intended as a way to more easily install existing extensions. So instead of finding extensions through something like https://github.com/directus-community/awesome-directus#extensions and installing them manually, you can click a button in the app to install it. This discussion however does not involve any new _types_ of extensions. Therefore, the _"[...] but do not provide a reasonable way to extend it"_ statement doesn't really apply to this discussion as far as I'm aware 🤔 When it comes to DX of extensions themselves, make sure to keep an eye on https://github.com/directus/directus/pull/15277 🙂

In any case, lets keep this thread constructive/productive and on topic and please be aware of [the repository's code of conduct](https://github.com/directus/directus/blob/main/code_of_conduct.md).

**Reply by Nitwel** [reply]
> we don't give any f**k about the needs of the users, we have the following profit-oriented agenda, which we target first, anything else - we don't care about it

🤣 "profit-oriented agenda" and Directus, yea tell me more jokes...
Did I maybe forgot to say that Directus is **100% free**? Yea that smells really profit oriented. 🤣 

> lol, again someone who hasn't try to build something reasonable with Directust but feels entitled to educate me....

Doesn't sound like you have build something with Directus either so I would suggest to you to do that first before complaining. 😉

**Reply by FullStackAlex** [reply]
@neerajklite, @Nitwel, @LasseRosenow 

> The website is showing for over one year „Coming Soon“. Unbelievable. Super irresponsible.
Even the marketplace gets published any time soon, your information policy on this killed all the fuss around Directus for me. Not to mention the wasted time of a couple of weeks on checking Directus out to develop a proper app.


Can you please point me to the exact place where I have been "rude", "offensive" or inappropriate in my argumentation (with quotes and adequate explanations)? I bet you can't. Obviously the simple fact of some open critics and legit express of disappointment is "rude" for you fanboys (and Dirctus' employees).

**Reply by dmatthams** [reply]
Someone make it stop

**Reply by FullStackAlex** [reply]
@Nitwel 
Don't get the reason for your hysterical laugh. You're ok? Need help?

> Did I maybe forgot to say that Directus is 100% free? Yea that smells really profit oriented.

Yes, you forgot to mention this. But either you're not experienced enough with modern IT or you're trying to mislead such who is inexperienced with it/IT by implying that free open source can't have any profit oriented agenda behind it.  

> Doesn't sound like you have build something with Directus either so I would suggest to you to do that first before complaining. 

Why does this sound to you?

**Reply by Nitwel** [reply]
So you are trying to tell me you didn't know that Directus was free while complaining about wasted time researching and looking into Directus? I think you might need to improve your research skills a bit as the front page of the GitHub repo clearly states that Directus is totally free. 😉

> implying that free open source can't have any profit oriented agenda behind it.

Oh I think you must have missunderstood me as I never said that. That's just what you interpreted into it.

![grafik](https://user-images.githubusercontent.com/22577866/187720877-0a46ef0d-5065-494d-8978-0e20c3574313.png)

**Reply by FullStackAlex** [reply]
@Nitwel
> Doesn't sound like you have build something with Directus either so I would suggest to you to do that first before complaining.

And I suggest you stop falsely spread misinformation about me. You surely can question how "proper" my attempted Directus app was. But I didn't provide any information about the attempted app, so stop claiming anything as a fact. Whereas dmatthams clearly says that he didn't even try to build anything serious with it yet (was just tinkering).

**Reply by FullStackAlex** [reply]
@Nitwel
> So you are trying to tell me you didn't know that Directus was 

No, I clearly said YOU forgot to say that it's free. I never claimed not to know that Directus is free. I barely can hold myself back to stay polite with you the way you implying how dumb I must be.

**Reply by FullStackAlex** [reply]
@Nitwel
just to be clear - do you work for Directus or are you just an independent contributer to the code base?

**Reply by rijkvanzanten** [reply]
@FullStackAlex This is turning into personal attacks, insults, and harassment. Please see [the code of conduct](https://github.com/directus/directus/blob/main/code_of_conduct.md) and refrain from further harmful conduct.

**Reply by FullStackAlex** [reply]
@rijkvanzanten 
can you please quote where I attack, insult or harass anyone? 
Also can you please explain why you didn't link/mention Nitwel and neerajklite in your comment they clearly attacked me personally (neerajklite tells me to STFU and Nitwel spreads false assumptions about me as facts) .

**Reply by LasseRosenow** [reply]
@Nitwel Actually I have to say I disagree with @FullStackAlex on almost everything they say, except that @FullStackAlex is not wrong in that open source projects can be used as advertisement for more profit oriented products, like the directus cloud services with some extra functionality even.

Though there is no information that the cloud services are supposed to be some "personal gain" and not to support the further development of the project itself, so I think @FullStackAlex accusations are just this. "Accusations". But they lack proof. But even if the cloud business does not fund the directus development, I think it is a really fair trade to get this software for free. Besides all this I wander why this even is of any importance, as far as I know the extension marketplace is not supposed to be cloud only haha.

**Reply by FullStackAlex** [reply]
@LasseRosenow 
Exactly, I just expressed my legit opinion. But the way I get attacked and harassed here speaks volumes about the Directus' community. Which, though, unfortunately isn't uncommon among open source projects backed by profit oriented companies.

**Reply by FullStackAlex** [reply]
@neerajklite
The lack of likes for your comment from Directus fanboys should be response enough from my side. 
But the lack of dislikes clearly shows the grade of hypocrisy and the true nature of this community.

**Reply by rijkvanzanten** [reply]
Based on the [Standards in the CoC](https://github.com/directus/directus/blob/main/code_of_conduct.md#our-standards), I consider the following deconstructive/harmful/unacceptable conduct in our public forums for various reasons, including being disrespectful, not accepting of criticism, not focussing on what's best for the overall community, not showing any empathy towards other members of the community, insulting comments, personal attacks, and public harassment:

> Unbelievable. Super irresponsible.

> Your, and your downvoter's feeling is wrong. You and your fanboys are hyping your platform as something as solid and complete. Also as far as I remember you raised this or last year a couple dozens of millions of bucks. "Open-Source" is not an excuse for attracting developers with tech buzz and then not to deliver essential features for ages. People invest valuable (life!!) time just to check out your promises!! Unlike you, I don't get paid by anyone for this.

> [...] Also downvoting critical comments personally is the lowest level :D

> [...] Otherwise I have to assume you all are Directus staff members considering how quickly you downvote.

> That's the problem. Next please

> [...] that probably will never come because it probably doesn't fit into the profit roadmap anymore.

> So if I criticize Ben or Directus, then I criticize "open source"?! Obviously wrong again.

> [...] But then just be transparent about it: we don't give any f**k about the needs of the users, we have the following profit-oriented agenda, which we target first, anything else - we don't care about it.

> [...] Ben and his fanboys are promising a great alternative to any other CMS but do not provide a reasonable way to extend it. That should be addressed in any hyped video and blog post. Ben is promising something but fails to deliver. Get it! [...]

> lol, again someone who hasn't try to build something reasonable with Directust but feels entitled to educate me....

> Can you please point me to the exact place where I have been "rude", "offensive" or inappropriate in my argumentation (with quotes and adequate explanations)? I bet you can't. [...]

> Don't get the reason for your hysterical laugh. You're ok? Need help?

> But either you're not experienced enough with modern IT or you're trying to mislead such who is inexperienced with it/IT by implying that free open source can't have any profit oriented agenda behind it.

> I barely can hold myself back to stay polite with you the way you implying how dumb I must be.

This user has now been blocked from interacting in the repo. Lets keep discussions constructive and for the benefit of the project, rather than hateful shouting matches around who's right or wrong and about what. Happy to discuss any of the points raised above in a more constructive manner in a separate thread.

**Comment by ghost** [comment]
I would recommend a model like that of drupal which is that the community develops and maintains the extensions/plugins/modules . A paid model like wordpress will promote no-collaboration multiple solutions for the same issue kinda ecosystem. This dilutes the quality and promotes a anti democratic system.

**Reply by rijkvanzanten** [reply]
I don't see a problem in allowing for both. Its up to the developer whether or not to charge for their time 👍🏻

**Reply by jdamboeck** [reply]
It is always up to the developer. With or without a marketplace for it. 

The marketplace is about promotion and accesability. What do you want to promote. 100 versions of a plugin or 1 working one. What will be better for the developer using it?

**Reply by hermesalvesbr** [reply]
100.

**Reply by jdamboeck** [reply]
So would like to spend your development time with research in which of those 100 does exactly does what you need and fits in your budget than implementing the one that just works?

100 devs doing 1 extension will get a better outcome than 100 devs doing a 100 imho.

**Reply by Michael-Nussbaumer** [reply]
Yeah and what if the 1 available solution not fits your usecase and/or budget? A kind of a rating system and stats how many times an extension is used can also help to filter for the better solutions. But I think nobody will stop devs to team up for maintaining some extensions.

**Reply by rijkvanzanten** [reply]
> 100 devs doing 1 extension will get a better outcome than 100 devs doing a 100 imho.

I disagree! Having multiple versions that solve the same problem drives competition and makes the whole ecosystem better because of it 🙂

**Reply by benhaynes** [reply]
How about we agree that 20 teams of 5 devs might be a good balance? 😄 Life is always about balance...

**Reply by rijkvanzanten** [reply]
Haha deal! 🤝

**Reply by kenlyle2** [reply]
wait...choice is aniti-democratic?  Looks like 1984 has arrived.

**Comment by eyecatchup** [comment]
@benhaynes I saw you created https://directus.market/, but it only says _coming soon_ and _track progess_. So, any progress/updates you can share?

**Reply by benhaynes** [reply]
Still a "Coming Soon" sort of thing. We have a few other (big) things we're getting out first... then this will be our main focus. We can't give formal time estimates, but I would say a few months is what we're aiming for.

**Reply by Dan6erbond** [reply]
I hate asking when something so awesome is provided for free, but what's the status on those "couple months"? I'm considering Directus for a project of mine and the market would make a huge difference.

**Reply by benhaynes** [reply]
See my response from yesterday... https://github.com/directus/directus/discussions/2820#discussioncomment-3797663

**Reply by mcqua007** [reply]
> Still a "Coming Soon" sort of thing. We have a few other (big) things we're getting out first... then this will be our main focus. We can't give formal time estimates, but I would say a few months is what we're aiming for.

@benhaynes  Hey Ben Thanks for all the hard work. I have been pushing Directus hard for a new company project & am so excited to use it. My team was blown away. I was wondering if there is a way to create custom extensions for cloud users, standard, community, & enterprise ? It appears we need to self serve which is fine, but I would love to pay for it to support all the hard work.

**Reply by benhaynes** [reply]
@mcqua007 — thank you!! Yes, Enterprise supports custom extensions, and we have Cloud, On-Prem, and Hybrid deployment options available in this tier too. Happy to answer any questions about that side of things, or you can shoot our Enterprise Team a message via our website. ❤️

**Reply by mcqua007** [reply]
@benhaynes I think I reached out regarding enterprise and pricing but have not heard back. for enterprise do you get ssh access to the server or how does that work ? BTW I have been a huge advocate for Directus for years! My boss wanted to go with Contentful and I was like NO we gotta use Directus. It is leagues above and the team behind it is amazing! (Just wanted you to know how thankful the community is!)

**Reply by benhaynes** [reply]
@mcqua007 — I've reached out to our team to ensure you get a response. 🙌

**Comment by Attacler** [comment]
So i want to be active in the marketplace and have a few questions:
 - Ben told me that there that subscriptions are going to be a thing, is this going to be seperated billing for month or year (like give 1 month away if someone pays for a year)
 - How much is the profit sharing (percentage) going to be?
 - Are payouts done trough creditnotes or how will it be handles (from an accounting standpoint)?
 - Can i provide standard collections? (ones that should be used in the extension)
     - Are users allowed to edit the data model of those fields?
     - Can we place collections under the (project) settings?
 - What would the development procedure look like?
    - Do we have to keep things in mind while making extensions? (for example how settings are managed or should we do it based on .env?)
    - How do we provide assets/code? for example, is it a .zip upload, a npm package (if so, how is the payment part handled?), or some other way?

**Reply by benhaynes** [reply]
All of this is still being figured out internally, but I've described our current intentions. If anyone else on our team has additional insight, that'd be great... but I want to be clear that anything can change before GA.

**Reply by Attacler** [reply]
> All of this is still being figured out internally, but I've described our current intentions. If anyone else on our team has additional insight, that'd be great... but I want to be clear that anything can change before GA.

Alright.
If i can be usefull in making decisions or joining discussions, let me know!

**Comment by u12206050** [comment]
Will extensions be able to be installed locally and committed to code and then deployed as part of a CI/CD pipeline? Or is the idea that all extensions install directly on the hosted instance?

**Reply by rijkvanzanten** [reply]
Either of those approaches will work @u12206050. The first approach would rely on the extensions folder for "local" committed code, the second approach can be done by relying on `npm` to distribute the extensions.

**Comment by hermesalvesbr** [comment]
Hello,

I'm going to make a suggestion to create a countdown, to calm our anxiety for this store.

A simple countdown on https://directus.market/ would give directus fans a big thumbs up.

Kind regards,

Hermes Souza

**Reply by LasseRosenow** [reply]
But isn't it, that they said so many times already, they don't want to give time estimates? Because its just tooo easy to delay and then would make customers angry for them not fulfilling their promises?

**Reply by hermesalvesbr** [reply]
That was two years ago, now it's another scenario and this can be revisited. Two years is a long time to not have answers.

Two years.

**Reply by benhaynes** [reply]
The update is that extension DX and the marketplace are the next big focus of our Engineering team. No concrete timing, but we're diving in in a few weeks, and then will give more updates and sneak peeks over time. Hopefully, we have something out in 6 months.... but timelines slip based on other priorities, which is why we don't try to predict them. 🙂

**Reply by hermesalvesbr** [reply]
Thanks for responding.

**Comment by DracoBlue** [comment]
An hint for those waiting for the market: there are three ways to find extensions (that I am aware of).

* Check awesome directus list at https://github.com/directus-community/awesome-directus#extensions
* Search npm for "directus-extension" https://www.npmjs.com/search?q=directus-extension
* Search github for "directus-extension" https://github.com/search?q=directus-extension&type=repositories

**Comment by fabiobeoni** [comment]
I have a question about the licensing model... 

Currently, due to the Directus GPL3 license, all marketplace plugins should also make the source code available open source. Does that fit with a marketplace?

Thanks.

**Reply by rijkvanzanten** [reply]
See https://github.com/directus/directus/discussions/17218 and lets discuss there 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by BlackDahlia313** [comment]
Question for anyone.

Was just talking about this while writing my guest blog and wondering how this will be persisted to new containers? Is the extension installed during bootstrap?

**Reply by paescuj** [reply]
Good question! Not sure if this has already been addressed. @Nitwel should be able to bring clarity.

**Reply by kevincam3** [reply]
I think they are loaded on bootstrap but don't quote me. I usually see a message about all the extensions that are being loaded everytime I start up my container. Extensions can also be reloaded automatically when they change if you enable that in the environment variable. https://docs.directus.io/self-hosted/config-options.html#extensions

**Reply by paescuj** [reply]
> I think they are loaded on bootstrap but don't quote me. I usually see a message about all the extensions that are being loaded everytime I start up my container. Extensions can also be reloaded automatically when they change if you enable that in the environment variable. https://docs.directus.io/self-hosted/config-options.html#extensions

Thank you for your input, @kevincam3! ❤️ You're right that extensions are loaded at the startup of Directus but I think @BlackDahlia313 was asking about how the extensions are persisted in a container after being installed via Marketplace, so that it is still available after a restart.

**Reply by rijkvanzanten** [reply]
@BlackDahlia313 Extensions are downloaded to the extensions folder and each container will read it from that volume so they're in sync 🙂

**Reply by rijkvanzanten** [reply]
Oh and we'll add in support to have these extensions saved to a storage adapter in case you want to use S3 instead of a volume https://github.com/directus/directus/issues/19803

**Reply by BlackDahlia313** [reply]
> Oh and we'll add in support to have these extensions saved to a storage adapter in case you want to use S3 instead of a volume #19803

Mic Drop. 

That answered everything. Very exciting stuff here! 

And Thanks @kevincam3  got a keen eye :)

**Reply by kevincam3** [reply]
@BlackDahlia313 & @paescuj Thanks for the love. I'm setting up our infrastructure to use Docker containers in production so the plan is any extensions from the Market we'll install and test them in a dev environment and them add them into the docker image before we deploy to production. I hope there's not something I'm missing here that could cause a problem but I'd like to keep my docker image repeatable. But if someone can predict a problem by all means let me know. I'm still a newbie when it comes to docker containers in production.

**Reply by BlackDahlia313** [reply]
@kevincam3 using a persistent volume or any external storage will keep each image repeatable as the extensions are loaded in this way for each new container that comes online. 

Its really clever and should make it pretty service agnostic. You should be all set :)

**Comment by Innovathor69** [comment]
Hey there,

any idea how to access this marketplace and extensions with the Railway-Deployment?

**Comment by eyecatchup** [comment]
😮 See, what just hit my inbox. It's happening! ❤️ 
 
![Bildschirmfoto 2024-02-27 um 17 58 13](https://github.com/directus/directus/assets/845131/48f5826d-35a5-4877-a185-352b7cf38d44)

**Reply by eyecatchup** [reply]
High res preview ([via](https://hs-20534155.f.hubspotemail.net/hub/20534155/hubfs/categories.png?upscale=true&name=categories.png))
![categories](https://github.com/directus/directus/assets/845131/1e917847-3b37-4967-9692-67c1f45f97b3)

PS: In case you wanna join the keynote presentation, here you go: https://lu.ma/leapweek
